src/runtime/virtcontainers/mount.go

```diff
@@ -226,6 +226,40 @@ func isDeviceMapper(major, minor int) (bool, error) {
 	return false, err
 }
 
+var blockFormatTemplates = map[string]string{
+	"dm":    "/sys/dev/block/%d:%d/dm",
+	"block": "/sys/dev/block/%d:%d/device/block",
+	"loop":  "/sys/dev/block/%d:%d/loop",
+}
+
+func isBlockDevice(major, minor int, formatTemplate string) (bool, error) {
+	sysPath := fmt.Sprintf(formatTemplate, major, minor)
+
+	_, err := os.Stat(sysPath)
+	if err == nil {
+		return true, nil
+	} else if os.IsNotExist(err) {
+		return false, nil
+	}
+
+	return false, err
+}
+
+func checkBlockDevice(major, minor int) (bool, error) {
+	var lastErr error = nil
+	for dev, template := range blockFormatTemplates {
+		isBD, err := isBlockDevice(major, minor, template)
+		if isBD {
+			mountLogger().Infof("Found storage driver for %s", dev)
+			return isBD, err
+		} else if err != nil {
+			lastErr = err
```

Comment:
Can we make this logic easier to understand like the following? I think if any error occurred while checking can just throw the error without continuing.

```
isBD, err := isBlockDevice(major, minor, template)
if err != nil {
    return 
}
```